py/torch_tensorrt/dynamo/conversion/impl/elementwise/base.py

```diff
@@ -148,7 +150,11 @@ def convert_binary_elementwise(
             ctx, rhs_val, trt_promoted_type, f"{name}_cast_rhs_val", target, source_ir
         )
 
-    if has_dynamic_shape(lhs_val.shape) or has_dynamic_shape(rhs_val.shape):
+    if len(lhs_val.shape) == len(rhs_val.shape) and all(
+        a == b or a == 1 or b == 1 for a, b in zip(lhs_val.shape, rhs_val.shape)
+    ):
+        logger.info(f"skip broadcast for {name}")
```

Comment:
Can you explain what's happening here ? If shape of lhs_val = (2, 3, 4) and rhs_val.shape = (2, 3, 1), we should not skip broadcast right ?